src/iexamine.cpp

```diff
@@ -2370,6 +2364,10 @@ void iexamine::fireplace( player &p, const tripoint &examp )
         if( has_bionic_firestarter ) {
             selection_menu.addentry( 2, true, 'b', _( "Use a CBM to start a fire" ) );
         }
+    } else if( p.weapon.damage_melee( DT_BASH ) ) {
+        selection_menu.addentry( 4, true, 's', _( "Put out fire" ) );
+    } else {
+        selection_menu.addentry( 4, false, 's', _( "Put out fire - wield bashing item" ) );
```

Comment:
Maybe something like
```suggestion
        selection_menu.addentry( 4, false, 's', _( "Put out fire (bashing item required)" ) );
```

Reply:
Excellent stuff!

Reply:
I deliberately want the player to choose an item by wielding it, rather then using random bashing item from inventory. Maybe it's less convenient but it seemed more natural to me.